streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -142,6 +170,20 @@ void openExisting(final ProcessorContext context) {
         return segments;
     }
 
+    List<Segment> allSegments() {
+        final List<Segment> segments = new ArrayList<>();
+        for (Segment segment : this.segments.values()) {
+            if (segment != null && segment.isOpen()) {
+                try {
+                    segments.add(segment);
+                } catch (InvalidStateStoreException ise) {
```

Comment:
This is not introduced in this PR, but I think this try / catch is actually buggy as it only wraps the `add` call which would never throw the exception, instead we should just throw exception when `segment.isOpen()` returns false (cc @dguy )

Note this comment is applicable to the original `List<Segment> segments(final long timeFrom, final long timeTo)` as well.

Also do not need the `null` check as well since we do not try to `getSegment(segmentId)` but access the segments list directly.